format/nes/nes.go

```diff
@@ -0,0 +1,322 @@
+	0: {Description: "RP2C02 (NTSC NES)"},
+	1: {Description: "RP2C07 (Licensed PAL NES)"},
+	2: {Description: "Multiple-region"},
+	3: {Description: "UA6538 (Dendy)"},
+}
+
+var vsPPUMapper = scalar.UintMap{
+	0:  {Description: "RP2C03B"},
+	1:  {Description: "RP2C03G"},
+	2:  {Description: "RP2C04-0001"},
+	3:  {Description: "RP2C04-0002"},
+	4:  {Description: "RP2C04-0003"},
+	5:  {Description: "RP2C04-0004"},
+	6:  {Description: "RC2C03B"},
+	7:  {Description: "RC2C03C"},
+	8:  {Description: "RC2C05-01 ($2002 AND $?? =$1B)"},
+	9:  {Description: "RC2C05-02 ($2002 AND $3F =$3D)"},
+	10: {Description: "RC2C05-03 ($2002 AND $1F =$1C)"},
+	11: {Description: "RC2C05-04 ($2002 AND $1F =$1B)"},
+	12: {Description: "RC2C05-05 ($2002 AND $1F =unknown)"},
+	13: {Description: "reserved"},
+	14: {Description: "reserved"},
+	15: {Description: "reserved"},
+}
+
+var vsHardwareMapper = scalar.UintMap{
+	0: {Description: "Vs. Unisystem (normal)"},
+	1: {Description: "Vs. Unisystem (RBI Baseball protection)"},
+	2: {Description: "Vs. Unisystem (TKO Boxing protection)"},
+	3: {Description: "Vs. Unisystem (Super Xevious protection)"},
+	4: {Description: "Vs. Unisystem (Vs. Ice Climber Japan protection)"},
+	5: {Description: "Vs. Dual System (normal)"},
+	6: {Description: "Vs. Dual System (Raid on Bungeling Bay protection)"},
+}
+
+var extConsoleMapper = scalar.UintMap{
+	0:  {Description: "[Regular NES/Famicom/Dendy]"},
+	1:  {Description: "[Nintendo Vs. System]"},
+	2:  {Description: "[Playchoice 10]"},
+	3:  {Description: "Regular Famiclone, but with CPU that supports Decimal Mode"},
+	4:  {Description: "Regular NES/Famicom with EPSM module or plug-through cartridge"},
+	5:  {Description: "V.R. Technology VT01 with red/cyan STN palette"},
+	6:  {Description: "V.R. Technology VT02"},
+	7:  {Description: "V.R. Technology VT03"},
+	8:  {Description: "V.R. Technology VT09"},
+	9:  {Description: "V.R. Technology VT32"},
+	10: {Description: "V.R. Technology VT369"},
+	11: {Description: "UMC UM6578"},
+	12: {Description: "Famicom Network System"},
+	13: {Description: "reserved"},
+	14: {Description: "reserved"},
+	15: {Description: "reserved"},
+}
+
+func decodeFileHeader(d *decode.D, nc *nesContext) {
+	// byte 0-3
+	d.FieldRawLen("identifier", 4*8, d.AssertBitBuf([]byte{0x4e, 0x45, 0x53, 0x1a})) // NES<EOF>
+
+	// byte 4
+	prgLSB := d.FieldU8("prg_rom_size_lsb")
+
+	// byte 5
+	chrLSB := d.FieldU8("chr_rom_size_lsb")
+
+	// byte 6
+	d.FieldU1("nametable_layout", scalar.UintMapSymStr{0: "vertical", 1: "horizontal"})
+	d.FieldBool("battery", scalar.BoolMapDescription{false: "Not present", true: "Present"})
+	trainer := d.FieldBool("trainer", scalar.BoolMapDescription{false: "Not present", true: "Present"})
+	d.FieldBool("alternative_nametables")
+	mapperLSB := d.FieldU4("mapper_lsb")
+
+	// byte 7 (half)
+	consoleType := d.FieldU2("console_type", consoleTypeMapper)
+	nes20 := d.FieldU2("nes20_identifier", scalar.UintMapDescription{0: "iNES", 1: "NES 2.0"})
```

Comment:
Maybe battery, trainer, console_type and nes20_identifier should have symbolic names and maybe sometimes a description if symbolic value is not descriptive enough?